File: gamepack/Decompiled/Username.java
public class Username implements Comparable {
   String name0;
   String cleanName0;

   public Username(String var1) {
      this.name0 = var1;
      this.cleanName0 = bk.aw_renamed(var1, LoginType.af);
   }

   public Username(String var1, LoginType var2) {
      this.name0 = var1;
      this.cleanName0 = bk.aw_renamed(var1, var2);
   }

   public String name() {
      return this.name0;
   }

   public String cleanName() {
      return this.cleanName0;
   }

   public boolean hasCleanName() {
      return this.cleanName0 != null;
   }

   public boolean equals(Object var1) {
      if (var1 instanceof Username) {
         Username var2 = (Username)var1;
         if (null == this.cleanName0) {
            return null == var2.cleanName0;
         } else if (var2.cleanName0 == null) {
            return false;
         } else {
            return this.hashCode() != var2.hashCode() ? false : this.cleanName0.equals(var2.cleanName0);
         }
      } else {
         return false;
      }
   }

   public int hashCode() {
      return null == this.cleanName0 ? 0 : this.cleanName0.hashCode();
   }

   public String toString() {
      return this.name();
   }

   public int compareTo0(Username var1) {
      if (this.cleanName0 == null) {
         return var1.cleanName0 == null ? 0 : 1;
      } else {
         return null == var1.cleanName0 ? -1 : this.cleanName0.compareTo(var1.cleanName0);
      }
   }

   public int compareTo(Object var1) {
      return this.compareTo0((Username)var1);
   }
}

File: find-names/src/main/java/mapper/identifiers/classes/Username.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.abstractclasses.OrderMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.predicateutilities.Predicate
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import mapper.wrappers.InstructionWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Opcodes
import org.objectweb.asm.Type
import org.runestar.client.common.startsWith

class Username : IdentityMapper.Class() {
    override val predicate = predicateOf<ClassWrapper> { it.interfaces.contains(Comparable::class.type) }
            .and { it.instanceFields.count() == 2 }
            .and { it.instanceFields.count { it.type == String::class.type } == 2 }

    class name0 : OrderMapper.InConstructor.Field(Username::class, 0) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == String::class.type }
        override val constructorPredicate = predicateOf<MethodWrapper> { it.arguments.size > 1 }
    }

    class cleanName0 : OrderMapper.InConstructor.Field(Username::class, 1) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == String::class.type }
        override val constructorPredicate = predicateOf<MethodWrapper> { it.arguments.size > 1 }
    }

    @MethodParameters()
    @DependsOn(name0::class)
    class name : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == String::class.type }
                .and { it.name == field<name0>().name }
    }

    @MethodParameters()
    @DependsOn(cleanName0::class)
    class cleanName : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == String::class.type }
                .and { it.name == field<cleanName0>().name }
    }

    @MethodParameters("other")
    class compareTo0 : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == Type.INT_TYPE }
                .and { it.arguments.startsWith(type<Username>()) }
    }

    @MethodParameters()
    class hasCleanName : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == Type.BOOLEAN_TYPE }
                .and { it.name != "equals" }
    }
}